packages/shared/src/accounts.ts

```diff
@@ -0,0 +1,3890 @@
+    return event.section === 'system' && event.method === 'ExtrinsicFailed'
+  })
+  assert(client.api.events.system.ExtrinsicFailed.is(errorEvent!.event))
+  const errorEventData = errorEvent!.event.data
+  assert(errorEventData.dispatchError.isToken)
+  const tokenError = errorEventData.dispatchError.asToken
+  expect(tokenError.isFrozen).toBeTruthy()
+}
+
+/**
+ * Test self-transfer of entire balance with `transfer_allow_death`.
+ *
+ * 1. Create Alice with 100 ED
+ * 2. Alice transfers 99 ED to herself
+ * 3. Verify that Alice is not reaped
+ * 4. Check events for exact numbers
+ */
+async function transferAllowDeathSelfTest<
+  TCustom extends Record<string, unknown> | undefined,
+  TInitStorages extends Record<string, Record<string, any>> | undefined,
+>(chain: Chain<TCustom, TInitStorages>) {
+  const [client] = await setupNetworks(chain)
+
+  const existentialDeposit = client.api.consts.balances.existentialDeposit.toBigInt()
+  const aliceBalance = existentialDeposit * 100n
+  const alice = await createAccountWithBalance(client, aliceBalance, '//fresh_alice')
+
+  expect(await isAccountReaped(client, alice.address)).toBe(false)
+
+  const transferAmount = existentialDeposit * 99n // Transfer almost everything to self
+  const transferTx = client.api.tx.balances.transferAllowDeath(alice.address, transferAmount)
+  await sendTransaction(transferTx.signAsync(alice))
+
+  await client.dev.newBlock()
+
+  // Get transaction fee
+  const events = await client.api.query.system.events()
+
+  const txPaymentEvent = events.find((record) => {
+    const { event } = record
+    return event.section === 'transactionPayment' && event.method === 'TransactionFeePaid'
+  })
+  assert(client.api.events.transactionPayment.TransactionFeePaid.is(txPaymentEvent!.event))
+  const txPaymentEventData = txPaymentEvent!.event.data
+  assert(txPaymentEventData.tip.toBigInt() === 0n, 'unexpected extrinsic tip')
+  expect(txPaymentEventData.actualFee.toBigInt()).toBeGreaterThan(0n)
+
+  // Alice should still be alive and balance only reduced by fees
+  expect(await isAccountReaped(client, alice.address)).toBe(false)
+  const aliceAccount = await client.api.query.system.account(alice.address)
+  expect(aliceAccount.data.free.toBigInt()).toBe(aliceBalance - txPaymentEventData.actualFee.toBigInt())
+
+  // Check no killing/creation related events occurred
+  const killedAccountEvent = events.find((record) => {
+    const { event } = record
+    return event.section === 'system' && event.method === 'KilledAccount'
+  })
+  expect(killedAccountEvent).toBeUndefined()
+
+  const newAccountEvent = events.find((record) => {
+    const { event } = record
+    return event.section === 'system' && event.method === 'NewAccount'
+  })
+  expect(newAccountEvent).toBeUndefined()
+
+  const dustLostEvent = events.find((record) => {
+    const { event } = record
+    return event.section === 'balances' && event.method === 'DustLost'
+  })
+  expect(dustLostEvent).toBeUndefined()
```

Comment:
this can be done by checkEvents toMatchSnapshot and expect the snapshot to be empty array